README.md

```diff
@@ -139,6 +193,81 @@ _Example macvlan CNI configuration_:
 4. nv-ipam plugin, as a result of CNI ADD command allocates a free IP from the IP Block of the
 corresponding IP Pool that was allocated for the node
 
+#### CIDRPool
+
+1. User (cluster administrator) defines a set of named CIDR Pools to be used for prefix allocation
+of container interfaces via CIDRPool CRD (more information in [Configuration](#configuration) section)
+
+_Example_:
+
+```yaml
+apiVersion: nv-ipam.nvidia.com/v1alpha1
+kind: CIDRPool
+metadata:
+  name: pool1
+  namespace: kube-system
+spec:
+  cidr: 192.168.0.0/16
+  gatewayIndex: 1
+  perNodeNetworkPrefix: 24
+  nodeSelector:
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: node-role.kubernetes.io/worker
+        operator: Exists
+
+```
+
+2. ipam-controller calculates and assigns unique subnets (with `perNodeNetworkPrefix` size) for each Node via CIDRPool Status:
+
+_Example_:
+
+```yaml
+apiVersion: nv-ipam.nvidia.com/v1alpha1
+kind: CIDRPool
+metadata:
+  name: pool1
+  namespace: kube-system
+spec:
+  cidr: 192.168.0.0/16
+  gatewayIndex: 1
+  perNodeNetworkPrefix: 24
+  nodeSelector:
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: node-role.kubernetes.io/worker
+        operator: Exists
+status:
+  allocations:
+  - gateway: 192.168.0.1
+    nodeName: host-a
+    prefix: 192.168.0.0/24
+  - gateway: 192.168.1.1
+    nodeName: host-b
+    prefix: 192.168.1.0/24
+
+```
+
+3. User specifies nv-ipam as IPAM plugin in CNI configuration
+
+_Example macvlan CNI configuration_:
+
+```json
+{
+  "type": "macvlan",
+  "cniVersion": "0.3.1",
+  "master": "enp3s0f0np0",
+  "mode": "bridge",
+  "ipam": {
+      "type": "nv-ipam",
+      "poolName": "pool1",
+      "poolType": "CIDRPool"
```

Comment:
I think the value should be lowercase here.

Reply:
done